File: BOJ/25000/25501.cpp
/**
 * 문제: 재귀의 귀재
 *
 * 시간 복잡도: O(N)
 * 공간 복잡도: O(N)
 *
 */

#include <bits/stdc++.h>
using namespace std;

int n, cnt;
char s[1004];

int recursion(const char *s, int l, int r) {
  cnt++;

  if (l >= r)
    return 1;
  else if (s[l] != s[r])
    return 0;
  else
    return recursion(s, l + 1, r - 1);
}

int isPalindrome(const char *s) { return recursion(s, 0, strlen(s) - 1); }

int main() {
  cin >> n;

  for (int i = 0; i < n; i++) {
    cin >> s;
    cout << isPalindrome(s) << " " << cnt << '\n';
    cnt = 0;
  }

  return 0;
};

File: BOJ/25000/25501.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

input.shift();
const arr = input;

let cnt = 0;

const recursion = (str, l, r) => {
  cnt++;

  if (l >= r) return 1;
  else if (str[l] !== str[r]) return 0;
  else return recursion(str, l + 1, r - 1);
};

const isPalindrome = (str) => {
  return recursion(str, 0, str.length - 1);
};

for (let it of arr) {
  console.log(isPalindrome(it), cnt);
  cnt = 0;
}
